test/e2e/shared/openstack.go

```diff
@@ -485,6 +485,42 @@ func GetOpenStackServers(e2eCtx *E2EContext, openStackCluster *infrav1.OpenStack
 	return srvs, nil
 }
 
+// GetOpenStackServer returns the server with the given ID along with the first
+// IP address it has in the OpenStackCluster network.
+func GetOpenStackServerWithIP(e2eCtx *E2EContext, id string, openStackCluster *infrav1.OpenStackCluster) (*ServerExtWithIP, error) {
```

Comment:
This is provider OpenStack, so maybe we can name this simply `GetServerWithIP`.

Reply:
Sure, but then we should also rename `GetOpenStackServers` above. I'm fine with either but would prefer to keep the same naming convention.

Reply:
Okay for me :)